runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkCachedArtifactPaths.java

```diff
@@ -0,0 +1,31 @@
+package org.apache.beam.runners.flink;
+
+/**
+ * Determines artifact path names within the
+ * {@link org.apache.flink.api.common.cache.DistributedCache}.
+ */
+public class FlinkCachedArtifactPaths {
+  private static final String DEFAULT_ARTIFACT_TOKEN = "default";
+
+  public static FlinkCachedArtifactPaths createDefault() {
+    return new FlinkCachedArtifactPaths(DEFAULT_ARTIFACT_TOKEN);
+  }
+
+  public static FlinkCachedArtifactPaths forToken(String artifactToken) {
+    return new FlinkCachedArtifactPaths(artifactToken);
+  }
+
+  private final String token;
+
+  private FlinkCachedArtifactPaths(String token) {
+    this.token = token;
+  }
+
+  public String getArtifactPath(String name) {
+    return String.format("ARTIFACT_%s_%s", token, name);
+  }
+
+  public String getManifestPath() {
+    return String.format("MANIFEST_%s", token);
```

Comment:
As above, this will need to be encoded if arbitrary. However, we are in charge of generating the token, so we can document a requirement on manifest tokens, such as ASCII alphanumeric only.